packages/material-ui/src/styles/useThemeProps.d.ts

```diff
@@ -1,25 +1,17 @@
-import { Components } from './components';
+import { Theme } from './createMuiTheme';
 
-export interface ThemeWithProps {
-  components?: Components;
-}
-
-export type ThemedProps<Theme, Name extends keyof any> = Theme extends {
+export type ThemedProps<T, Name extends keyof any> = T extends {
   components: Record<Name, { defaultProps: infer Props }>;
 }
   ? Props
   : {};
 
-export interface AdditionalThemeProps<Theme> {
+export interface AdditionalThemeProps<T> {
```

Comment:
Please revert these changes. Single letter type parameters are reserved for truly generic values. This type parameter has actual semantics: Some theme.

Reply:
@eps1lon  are you talking about changes like this ?

```
import { Theme } from './createMuiTheme';

export type ThemedProps<Theme, Name extends keyof any> = Theme extends {
  components: Record<Name, { defaultProps: infer Props }>;
}
  ? Props
  : {};

export interface AdditionalThemeProps<Theme> {
  isRtl: boolean;
  theme: Theme;
}

export default function useThemeProps<T = Theme, Props = {}, Name extends keyof any = any>(params: {
  props: Props;
  name: Name;
}): Props & ThemedProps<T, Name> & AdditionalThemeProps<T>;
```

Reply:
I'm talking about

```diff
-Theme
+T
```

These reduce readability.

Reply:
But let's not concentrate on that for now. It's more important to understand why these changes should happen.

Reply:
> Thanks for the work.
> 
> Could you explain what behavior you change with the type and runtime changes? Both preferably with a test.

Actually functionally did change because it's only fixed for typescript typings in comment https://github.com/mui-org/material-ui/pull/24383#discussion_r556837791. Because all other components are in *.js files you can't see that useThemeProps returns the same theme from the context but in the typescript component, you will get a compile error that is not the same theme type.

Reply:
> Actually functionally did not change because it's only fix for typescript typings in comment #24383 (comment). 

But you did change a `.js` file. If that doesn't change any functionality, could you please revert it?

> And I don't know even if I can test typescript types in this project or you got some advice about how to do it?

You can check existing `.spec.tsx?` files. For checking that code is rejected by the type-checker you can use [`@ts-expect-error`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments). For checking that a particular type is returned you can use `expectType` from `@material-ui/types`. For checking that code is accepted, just add the code. If if it's rejected by the type-checker, compilation will fail

Reply:
@eps1lon Ok maybe we can close this pull request. Can I convert useThemeProps and getThemeProps to typescript, because the functionally have similar typing even and both are incorrect? getThemeProps missing actual theme property and so on.
Provided typings do not reflect functionality useThemeProps and getThemeProps provides.